build.gradle

```diff
@@ -37,6 +37,8 @@ subprojects { project ->
                     "-Xlint:unchecked",
                     "-Xlint:rawtypes",
                     "-Werror",
+                    // don't analyze generated code; AutoValue code fails UnnecessaryParentheses check
+                    "-XepExcludedPaths:.*/build/classes/.*",
```

Comment:
Any reason for not using `-XepDisableWarningsInGeneratedCode` instead?

Also, does that mean that AutoValue-generated **sources** end up into `build/classes`? You may want to set `options.annotationProcessorGeneratedSourcesDirectory` on `JavaCompile` tasks to put them somewhere else so they're not packaged into the JAR. Fwiw, `net.ltgt.apt` will do that for you :wink: 
(I checked, and indeed nulaway-0.6.0.jar contains AUtoValue-generated sources)

Reply:
We need this since we use `-Werror`, and `-XepDisableWarningsInGeneratedCode` only disables warnings.

Nice tip on sources ending up in the final artifact!  I opened #245 on fixing.

Reply:
`-XepDisableWarningsInGeneratedCode` happens before warnings are emitted, so `-Werror` won't change the outcome (vs excluded paths), and it helps catch Error Prone errors.